src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,314 @@
+
+    private fun removeCommand() = CommandBuilder
+        .begin("remove")
+        .parameter(
+            ParameterBuilder
+                .begin<String>("name")
+                .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                .optional()
+                .build()
+        )
+        .handler { command, args ->
+            checkInGame()
+
+            if (fakePlayers == null || fakePlayers!!.isEmpty()) {
+                throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+            }
+
+            val name = args.getOrNull(0)?.toString() ?: "FakePlayer"
+            var anyRemoved = false
+
+            fakePlayers?.removeIf { fakePlayer ->
+                @Suppress("ReplaceCallWithBinaryOperator")
+                val remove = name.equals(fakePlayer.name.string)
+                if (remove) {
+                    world.removeEntity(fakePlayer.id, Entity.RemovalReason.KILLED)
+                    chat(
+                        regular(
+                            command.result(
+                                "fakePlayerRemoved",
+                                BigDecimal(fakePlayer.x).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.y).setScale(1, RoundingMode.HALF_UP).toDouble(),
+                                BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
+                            )
+                        )
+                    )
+                    anyRemoved = true
+                    return@removeIf true
+                } else {
+                    return@removeIf false
+                }
+            }
+
+            if (!anyRemoved) {
+                chat(warning((command.result("noFakePlayerNamed", name))))
+                chat(regular(command.result("currentlySpawned")))
+                fakePlayers!!.forEach { fakePlayer ->
+                    chat(regular("- " + fakePlayer.name.string))
+                }
+            }
+        }
+        .build()
+
+    private fun clearCommand() = CommandBuilder
+        .begin("clear")
+        .handler { _, _ ->
+            checkInGame()
+
+            if (fakePlayers == null || fakePlayers!!.isEmpty()) {
+                throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+            }
+
+            fakePlayers!!.removeIf { fakePlayer ->
+                world.removeEntity(fakePlayer.id, Entity.RemovalReason.DISCARDED)
+                return@removeIf true
+            }
+            fakePlayers = null
+        }
+        .build()
+
+    @Suppress("SpellCheckingInspection")
+    private fun startRecordingCommand() = CommandBuilder
+        .begin("startrecording")
+        .handler { command, _ ->
+            if (recording) {
+                throw CommandException(command.result("alreadyRecording"))
+            }
+
+            recording = true
+            snapshots = LinkedList<PosPoseSnapshot>()
```

Comment:
_Never_ use a linked list if you don't have a very good reason to do so. There are close to zero use cases for linked list and a simple `ArrayList` is almost always the better option ;)

If you are interested in which such use cases could be [this](https://stackoverflow.com/questions/670104/what-are-real-world-examples-of-when-linked-lists-should-be-used) StackOverflow thread lists all of them.